library/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java

```diff
@@ -1453,6 +1502,10 @@ public void initializeOutput(ExtractorOutput output, int trackId) throws ParserE
         case CODEC_ID_SUBRIP:
           mimeType = MimeTypes.APPLICATION_SUBRIP;
           break;
+        case CODEC_ID_ASS:
+          mimeType = MimeTypes.TEXT_SSA;
+          ssaHeader = codecPrivate;
```

Comment:
OK, I can get the header into the format object and I can see it is there in the decoder factory createDecoder call in onStreamChanged in the TextRenderer.

But the decode method of the decoder is called from the decoder thread, not from the text. 

If I plumb the format data into the decoder from the TextRenderer doesn't that make the decoder stateful again? The subtitle objects are created in the decoder thread and i can't see any way of passing Format data into each call to the decoder that would make it stateless.

The styles need to be accessible in the decoder at decode time if I'm going to create the spans at the time the cues are created.